### PR TITLE
chore(android): upgrade to latest gradle version + add verify script

### PR DESCRIPTION
**Pull request by Oupsla**
Hello 👋 

What I have done in this PR : 

- Upgrade gradle to the latest version (7.2)
- Remove buildToolsVersion cause there is now default version with gradle (so no need to inject it)
- Add verify scripts (coming from capacitor repo)

I have done this cause building my capacitor project with your plugin returns me an error about the buildToolsVersion : 

```
The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (30.0.2) for Android Gradle Plugin 4.2.1.
Android SDK Build Tools 30.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,11 @@
     "build": "npm run clean && tsc && rollup -c rollup.config.js",
     "clean": "rimraf ./dist",
     "watch": "tsc --watch",
-    "prepare": "npm run build"
+    "prepare": "npm run build",
+    "verify": "npm run verify:ios && npm run verify:android && npm run verify:web",
+    "verify:ios": "cd ios && pod install && xcodebuild -workspace Plugin.xcworkspace -scheme Plugin -destination \"platform=iOS Simulator,name=iPhone 12\" && cd ..",
```
Comment (tafelnl) [inline package.json:41]: What is the purpose of `-destination \"platform=iOS Simulator,name=iPhone 12\"` here? In the official Push Notifications Plugin of Capacitor, this part is missing. See: https://github.com/ionic-team/capacitor-plugins/blob/main/push-notifications/package.json#L32

**Review by tafelnl — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! Looks good to me. Although I have one question, which I noted in this review.

File: package.json (hunk 1 above)
Comment (Oupsla) [reply package.json:41]: I add the target to Iphone 12, cause without it, xcode (on command line) will not follow the minimum platform you specify in your Podfile and you will get errors like `is only available in Mac Catalyst 14.0 or newer`

Comment (pwespi) [reply package.json:41]: I had a similar issue on my plugin (bluetooth-le) and used the same solution. I also opened a PR to add this to the plugin template: https://github.com/ionic-team/create-capacitor-plugin/pull/51.

In that PR I also added `xcodebuild clean build test ...` such that unit tests are run with `verify:ios`.

Comment (tafelnl) [reply package.json:41]: Thanks for clarifying guys. Seems like a legit use case. So I'll go ahead and merge it now!

**Pull request merged**